R2API/ColorsAPI.cs

```diff
@@ -0,0 +1,140 @@
+            return newIndex;
+        }
+
+        /// <summary>
+        /// Adds a new DamageColor using a <see cref="SerializableDamageColor"/>
+        /// <para>The <see cref="DamageColorIndex"/> is set in <paramref name="serializableDamageColor"/>'s <see cref="SerializableDamageColor.DamageColorIndex"/> property</para>
+        /// </summary>
+        /// <param name="serializableDamageColor">The <see cref="SerializableDamageColor"/> to add</param>
+        public static void AddSerializableDamageColor(SerializableDamageColor serializableDamageColor) {
+            if (!Loaded)
+                throw new InvalidOperationException($"{nameof(ColorsAPI)} is not loaded. Please use [{nameof(R2APISubmoduleDependency)}({nameof(ColorsAPI)})]");
+
+            if (addedSerializableDamageColors.Contains(serializableDamageColor)) {
+                R2API.Logger.LogError($"Attempted to add SerializableDamageColor {serializableDamageColor} twice! aborting.");
+                return;
+            }
+
+            serializableDamageColor.DamageColorIndex = RegisterDamageColor(serializableDamageColor.color);
+        }
+        #endregion
+
+        #region Color Catalog Public Methods
+        /// <summary>
+        /// Adds a new Color to the game's <see cref="ColorCatalog"/>
+        /// </summary>
+        /// <param name="color">The color for the new Color for the <see cref="ColorCatalog"/></param>
+        /// <returns>The <see cref="ColorCatalog.ColorIndex"/> associated with <paramref name="color"/></returns>
+        /// <exception cref="InvalidOperationException"></exception>
+        public static ColorCatalog.ColorIndex RegisterColor(Color color) {
+            if (!Loaded)
+                throw new InvalidOperationException($"{nameof(ColorsAPI)} is not loaded. Please use [{nameof(R2APISubmoduleDependency)}({nameof(ColorsAPI)})]");
+
+            int nextColorIndex = ColorCatalog.indexToColor32.Length;
+            ColorCatalog.ColorIndex newIndex = (ColorCatalog.ColorIndex)nextColorIndex;
+            HG.ArrayUtils.ArrayAppend(ref ColorCatalog.indexToColor32, color);
+            HG.ArrayUtils.ArrayAppend(ref ColorCatalog.indexToHexString, Util.RGBToHex(color));
+            return newIndex;
+        }
+
+        /// <summary>
+        /// Adds a new Color to the <see cref="ColorCatalog"/> using a <see cref="SerializableColorCatalogEntry"/>
+        /// <para>The <see cref="ColorCatalog.ColorIndex"/> is set in <paramref name="serializableColor"/>'s <see cref="SerializableColorCatalogEntry.ColorIndex"/></para>
+        /// </summary>
+        /// <param name="serializableColor">The <see cref="SerializableDamageColor"/> to add</param>
+        public static void AddSerializableColor(SerializableColorCatalogEntry serializableColor) {
+            if (!Loaded)
+                throw new InvalidOperationException($"{nameof(ColorsAPI)} is not loaded. Please use [{nameof(R2APISubmoduleDependency)}({nameof(ColorsAPI)})]");
```

Comment:
minor thing but could use a ThrowIfNotLoaded method there for reducing code copy paste

Reply:
I am suprised an utility method for this doesnt exist.

Then again, this pattern is present on every single API so it doesnt matter much at this point, esppecially since the splitting should make this not necesary.